lib/utils/getOverlayEntry.js

```diff
@@ -0,0 +1,57 @@
+const querystring = require('querystring');
+
+/**
+ * Creates an object that contains two entry arrays: the prependEntries and overlayEntries
+ * @param {import('../types').NormalizedPluginOptions} options Configuration options for this plugin.
+ * @param {import('webpack').Compiler.options.devServer | undefined} devServer The webpack devServer config
+ * @returns {import('./injectRefreshEntry').AdditionalEntries} An object that contains the Webpack entries for prepending and the overlay feature
```

Comment:
Minor thing - I think we should move the type definition here instead of having it in `injectRefreshEntry` for two reasons:

1. This would prevent recursive imports
2. This would conceptually imply what `injectRefreshEntry` consumes is utterly dependent on this function's implementation, but not the other way around.

Reply:
Then in `injectRefreshEntry` we could just either import the type and use it, or just do `ReturnType<import('./injectRefreshEntry')>`.